Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverridePublicHandler.cpp

```diff
@@ -50,16 +70,15 @@ namespace AzToolsFramework
             // redundant checks Eg: "Instances/InstanceB/....' . So we skip the first 2 tokens here.
             if (focusedInstance.has_value() && absoluteEntityAliasPath.size() > 2)
             {
-                AZStd::string_view overriddenInstanceKey = absoluteEntityAliasPath[1].GetKey().GetStringView();
-                InstanceOptionalReference overriddenInstance = focusedInstance->get().FindNestedInstance(overriddenInstanceKey);
-                if (overriddenInstance.has_value())
+                AZStd::string_view topMostInstanceKey = absoluteEntityAliasPath[1].GetKey().GetStringView();
+                InstanceOptionalReference topMostInstance = focusedInstance->get().FindNestedInstance(topMostInstanceKey);
+                if (topMostInstance.has_value())
                 {
                     auto pathIterator = absoluteEntityAliasPath.begin() + 2;
-                    AZ::Dom::Path modifiedPath(pathIterator, absoluteEntityAliasPath.end());
-                    return m_prefabOverrideHandler.AreOverridesPresent(modifiedPath, overriddenInstance->get().GetLinkId());
+                    return AZStd::pair(AZ::Dom::Path(pathIterator, absoluteEntityAliasPath.end()), topMostInstance->get().GetLinkId());
```

Comment:
I think I tried doing the similar thing before, but eventually I ended up calling ClimbUp first and GetRelativePath from climbed instances and then retrieve the link id from the climbed instances.

And can this logic be moved inside InstanceToTemplatePropagator?

Reply:
I don't think this belongs in the InstanceToTemplatePropagator because it tries to do something very specific with the fetched path, i.e removing the first 2 path tokens and fetching the linkId from there. I think we should avoid polluting the InstanceToTemplatePropagator with a lot of small helper functions like this. If it turns out that other override workflows also need this, we can revisit this then. But for now, I think it's best to have util/helper functions as private members instead of public interface methods.